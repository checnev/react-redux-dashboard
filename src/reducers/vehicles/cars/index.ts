import {
  GET_CARS_START,
  GET_CARS_SUCCESS,
  GET_CARS_FAIL,
  CarsActionTypes,
} from 'actions/vehicles/cars';

export enum CarStatus {
  Free,
  Busy,
  Repair,
}

export interface Car {
  id: number;
  stateNumber: string;
  model: string;
  status: CarStatus;
  color: string;
  imageUrl: string;
  coordinats: number[];
}

export interface CarsState {
  cars: Car[];
  isFetching: boolean;
}

const initialState: CarsState  = {
  cars: [],
  isFetching: false,
};

const carsReducer = (
  state = initialState,
  action: CarsActionTypes
): CarsState => {
  switch (action.type) {
    case GET_CARS_START:
      return { ...state, isFetching: true };
    case GET_CARS_SUCCESS: 
      return {
        ...state,
        isFetching: false,
        cars: action.payload,
      };
    case GET_CARS_FAIL:
      return {
        ...state,
        isFetching: false,
        cars: [],
      };
    default:
      return { ...state };
  }
};

export default carsReducer;
