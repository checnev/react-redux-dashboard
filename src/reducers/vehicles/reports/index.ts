import {
  GET_CAR_REPORTS_START,
  GET_CAR_REPORTS_SUCCESS,
  GET_CAR_REPORTS_FAIL,
  CarReportsActionTypes
} from 'actions/vehicles/reports';

import type { Moment } from 'moment';
import type { Car } from 'reducers/vehicles/cars';

export interface CarReport {
  income: number;
  incidents: number;
  countOfLeases: number;
}

export interface CarDailyReport extends CarReport {
  date: Moment;
  carId: number;
}

export type CarReports = WeakMap<Car, ReadonlyArray<CarDailyReport>>;

export interface ReportsState {
  reports: CarReports | null;
  isFetching: boolean;
  requestId: string | null;
}

const initialState: ReportsState = {
  reports: null,
  isFetching: false,
  requestId: null,
};

const reportsReducer = (
  state = initialState,
  action: CarReportsActionTypes
): ReportsState => {

  switch (action.type) {
    case GET_CAR_REPORTS_START:
      return { 
        ...state,
        isFetching: true,
        requestId: action.requestId,
      };
    case GET_CAR_REPORTS_SUCCESS:
      if (state.requestId !== action.requestId) return state;
      return {
        ...state,
        isFetching: false,
        reports: action.payload,
        requestId: null,
      };
    case GET_CAR_REPORTS_FAIL:
      if (state.requestId !== action.requestId) return state;
      return {
        ...state,
        isFetching: false,
        reports: null,
        requestId: null,
      };
    default:
      return { ...state };
  }
};

export default reportsReducer;
