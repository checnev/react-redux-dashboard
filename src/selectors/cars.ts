import { createSelector } from 'reselect';
import type { RootState } from 'store';
import type { Car } from 'reducers/vehicles/cars';
import type { CarReport, CarReports } from 'reducers/vehicles/reports';

export interface CarWithReport extends Car {
  report?: CarReport;
}

export const carsSelector = (state: RootState) => state.vehicles.cars.cars;
export const reportsSelector = (state: RootState) => state.vehicles.reports.reports;

export const carWithReportsSelector = createSelector(
  carsSelector,
  reportsSelector,
  (cars: Car[], reports: CarReports | null) => {
    const carList: CarWithReport[] = [];

    for (let car of cars) {
      const carReports = reports?.get(car);
      const carData: CarWithReport = Object.assign({}, car);

      if (carReports) {
        carData.report = carReports.reduce((summary: CarReport, report) => {
          summary.income += report.income;
          summary.countOfLeases += report.countOfLeases;
          summary.incidents += report.incidents;

          return summary;
        }, {
          income: 0,
          countOfLeases: 0,
          incidents: 0,
        });
      }

      carList.push(carData);
    }

    return carList;
  }
);
