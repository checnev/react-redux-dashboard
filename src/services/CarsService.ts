import moment, { Moment } from 'moment';
import type { Car } from 'reducers/vehicles/cars';
import type { CarDailyReport, CarReports } from 'reducers/vehicles/reports';


class CarsService {
  private static cars: Car[] = [
    {
      id: 1,
      stateNumber: 'С 012 РС',
      model: 'Hyndai Solaris (2014)',
      status: 0,
      color: '#d76830',
      imageUrl: '/images/cars/hyundai-solaris-2014.jpg',
      coordinats: [56.835832, 60.615666],
    },
    {
      id: 2,
      stateNumber: 'М 123 ВС',
      model: 'Kia Rio (2014)',
      status: 1,
      color: '#808483',
      imageUrl: '/images/cars/kia-rio-2014.jpg',
      coordinats: [56.816553, 60.564730],
    },
    {
      id: 3,
      stateNumber: 'О 684 РТ',
      model: 'Ford Mondeo (2018)',
      status: 2,
      color: '#655c57',
      imageUrl: '/images/cars/ford-mondeo-2018.jpg',
      coordinats: [56.831751, 60.562233],
    },
  ];

  public static getCarById(id: string): Promise<Car | undefined> {

    return new Promise((resolve) => {
      setTimeout(() => {
        const car = CarsService.cars.find((car) => String(car.id) === id);
        resolve(car);
      }, 2500);
    });
  }
  public static getCars(): Promise<Car[]> {
    return new Promise((resolve) => {
      setTimeout(() => resolve(CarsService.cars), 2500);
    });
  }

  public static getCarReports(cars: Car[], from: Moment, to: Moment): Promise<CarReports> {
    const carsData: CarReports = new WeakMap();
    const days = Math.abs(to.diff(from, 'd'));
    
    cars.forEach((car) => {
      const data: CarDailyReport[] = Array.from(
        { length: days }, 
        (day, index) => ({
          carId: car.id,
          date: moment(from).add(index, 'd'),
          income: Math.floor(Math.abs(Math.cos(index + car.id) * 50000)),
          countOfLeases: 10 + index + car.id,
          incidents: Math.floor(index * 0.5) % car.id,
        })
      );
      carsData.set(car, data);
    });

    return new Promise((resolve) => {
      setTimeout(() => resolve(carsData), 2500);
    });
  }
}

export default CarsService;
