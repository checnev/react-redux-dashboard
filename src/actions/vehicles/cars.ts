import CarsService from 'services/CarsService';
import { push, CallHistoryMethodAction } from 'connected-react-router';

import type { RootState } from 'store';
import type { ThunkAction } from 'redux-thunk';
import type { Car } from 'reducers/vehicles/cars';

interface GetCarsStartAction {
  type: typeof GET_CARS_START;
}

interface GetCarsSuccessAction {
  type: typeof GET_CARS_SUCCESS;
  payload: Car[];
}

interface GetCarsFailAction {
  type: typeof GET_CARS_FAIL;
  payload: string;
}

export type CarsActionTypes = GetCarsStartAction 
  | GetCarsSuccessAction 
  | GetCarsFailAction
  | CallHistoryMethodAction;

export const GET_CARS_START = 'GET_CARS_START';
export const GET_CARS_SUCCESS = 'GET_CARS_SUCCESS';
export const GET_CARS_FAIL = 'GET_CARS_FAIL';

export const getCarById = (id: string): ThunkAction<void, RootState, unknown, CarsActionTypes> => (
  async (dispatch) => {
    dispatch({
      type: GET_CARS_START,
    });

    try {
      const car = await CarsService.getCarById(id);

      if (!car) throw new Error(`Car with id ${id} doesn't exist.`);

      dispatch({
        type: GET_CARS_SUCCESS,
        payload: [car],
      });
    } catch (err) {
      dispatch({
        type: GET_CARS_FAIL,
        payload: err.message,
      });

      dispatch(push('/404'));
    }
  }
);

export const getCars = (): ThunkAction<void, RootState, unknown, CarsActionTypes> => (
  async (dispatch) => {
    dispatch({
      type: GET_CARS_START,
    });

    try {
      const cars = await CarsService.getCars();
      dispatch({
        type: GET_CARS_SUCCESS,
        payload: cars,
      });
    } catch (err) {
      dispatch({
        type: GET_CARS_FAIL,
        payload: err.message,
      });
    }
  }
);
