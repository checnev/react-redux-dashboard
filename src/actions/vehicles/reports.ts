import CarsService from 'services/CarsService';
import { v4 as uuid4 } from 'uuid';

import type { Moment } from 'moment';
import type { RootState } from 'store';
import type { ThunkAction } from 'redux-thunk';
import type { Car } from 'reducers/vehicles/cars';
import type { CarReports } from 'reducers/vehicles/reports';

interface ActionWithRequestId {
  requestId: string | null;
}

interface GetCarReportsStartAction extends ActionWithRequestId {
  type: typeof GET_CAR_REPORTS_START;
}

interface GetCarReportsSuccessAction extends ActionWithRequestId {
  type: typeof GET_CAR_REPORTS_SUCCESS;
  payload: CarReports;
}

interface GetCarReportsFailAction extends ActionWithRequestId {
  type: typeof GET_CAR_REPORTS_FAIL;
  payload: string;
}

export type CarReportsActionTypes = GetCarReportsStartAction | GetCarReportsSuccessAction | GetCarReportsFailAction;

export const GET_CAR_REPORTS_START = 'GET_CAR_REPORTS_START';
export const GET_CAR_REPORTS_SUCCESS = 'GET_CAR_REPORTS_SUCCESS';
export const GET_CAR_REPORTS_FAIL = 'GET_CAR_REPORTS_FAIL';

export const getCarReports = (
  cars: Car[],
  from: Moment,
  to: Moment
): ThunkAction<void, RootState, unknown, CarReportsActionTypes> => (
  async (dispatch) => {
    const requestId = uuid4();
    dispatch({
      type: GET_CAR_REPORTS_START,
      requestId,
    });
    
    try {
      const data = await CarsService.getCarReports(cars, from, to);
      dispatch({
        type: GET_CAR_REPORTS_SUCCESS,
        payload: data,
        requestId,
      });
    } catch (err) {
      dispatch({
        type: GET_CAR_REPORTS_FAIL,
        payload: err.message,
        requestId,
      });
    }
  }
);
