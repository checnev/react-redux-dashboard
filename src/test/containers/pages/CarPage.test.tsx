import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { waitFor, screen } from '@testing-library/react';
import pretty from 'pretty';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import state from 'test/data/initialState';
import CarsService from 'services/CarsService';
import { createCarReports } from 'test/utils/reports';
import cars from 'test/data/cars';
import CarPageContainer from 'containers/pages/Car';

const getCars = jest.fn().mockResolvedValue(cars);
const getCarReports = jest.fn().mockResolvedValue(
  createCarReports(cars, state.dashboard.from, state.dashboard.to)
);

CarsService.getCars = getCars;
CarsService.getCarReports = getCarReports;


jest.mock('react-router-dom', () => {
  const utils = jest.requireActual('react-router-dom');
  return {
    ...utils,
    useParams: () => ({ id: '1' }),
  };
});

jest.mock('react-yandex-maps', () => ({
  YMaps: (props: any) => <div data-testid="YMaps" {...props}></div>,
  Map: (props: any) => {
    const { defaultState, instanceRef, ...otherProps } = props;
    return <div data-testid="Map" {...otherProps}></div>
  },
  Placemark: (props: any) => <div data-testid="Placemark" {...props}></div>,
}));

jest.mock('recharts', () => ({
  AreaChart: () => <div data-testid="AreaChart"></div>,
  Area: () => <div data-testid="area"></div>,
  Tooltip: () => <div data-testid="Tooltip"></div>,
  XAxis: () => <div data-testid="XAxis"></div>,
  YAxis: () => <div data-testid="YAxis"></div>,
  CartesianGrid: () => <div data-testid="CartesianGrid"></div>,
  ResponsiveContainer: () => <div data-testid="ResponsiveContainer"></div>,
}));

jest.mock('@material-ui/core/TablePagination', () => {
  return function DummyTablePagination() {
    return <div data-testid="pagination"></div>;
  };
});


const mockStore = configureMockStore([thunk]);
let store: any = null;

let container: any = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);

  store = mockStore(state);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('CarPageContainer', () => {

  const setUp = (component: React.ReactNode) => (
    <Provider store={store}>
      <Router>
        {component}
      </Router>
    </Provider>
  );

  it('CarPageContainer render', async () => {
    await act(async () => {
      render(setUp(<CarPageContainer />), container);
    });

    await waitFor(() => {
      expect(screen.queryByDisplayValue('Hyundai solaris 2014')).toBeDefined();
    });

    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot();
  });
  
});
