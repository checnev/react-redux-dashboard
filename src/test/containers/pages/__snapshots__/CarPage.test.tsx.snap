// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarPageContainer CarPageContainer render 1`] = `
"<div class=\\"MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-3\\">
  <div class=\\"MuiGrid-root makeStyles-header-2 MuiGrid-item MuiGrid-grid-xs-12\\">
    <h1 class=\\"MuiTypography-root makeStyles-h1-3 MuiTypography-h4 MuiTypography-colorPrimary\\">Hyundai solaris 2014</h1>
    <div class=\\"makeStyles-dateRange-4\\">
      <div class=\\"makeStyles-root-9\\">
        <div class=\\"MuiFormControl-root MuiTextField-root makeStyles-input-10\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">From</label>
          <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" name=\\"from\\" type=\\"date\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"2020-12-10\\"></div>
        </div>
        <div class=\\"MuiFormControl-root MuiTextField-root makeStyles-input-10\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">To</label>
          <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" name=\\"to\\" type=\\"date\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"2020-12-17\\"></div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\">
    <hr class=\\"MuiDivider-root\\">
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\">
    <div class=\\"MuiPaper-root makeStyles-paper-6 MuiPaper-elevation1 MuiPaper-rounded\\"><img src=\\"/images/solaris.jpg\\" alt=\\"Hyundai solaris 2014\\" class=\\"makeStyles-carImage-5\\"></div>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\">
    <div class=\\"MuiPaper-root makeStyles-paper-6 MuiPaper-elevation1 MuiPaper-rounded\\">
      <dl class=\\"MuiGrid-root makeStyles-root-11 MuiGrid-container MuiGrid-spacing-xs-2\\">
        <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
          <dt class=\\"MuiTypography-root MuiTypography-body1\\">Модель:</dt>
        </div>
        <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
          <dd class=\\"MuiTypography-root MuiTypography-body1\\">Hyundai solaris 2014</dd>
        </div>
        <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
          <dt class=\\"MuiTypography-root MuiTypography-body1\\">Гос. Номер:</dt>
        </div>
        <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
          <dd class=\\"MuiTypography-root MuiTypography-body1\\">A 100 AA</dd>
        </div>
        <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
          <dt class=\\"MuiTypography-root MuiTypography-body1\\">Цвет:</dt>
        </div>
        <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
          <dd class=\\"MuiTypography-root MuiTypography-body1\\">
            <div class=\\"makeStyles-root-13\\" style=\\"width: 30px; height: 10px; background-color: rgb(255, 255, 255);\\"></div>
          </dd>
        </div>
        <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
          <dt class=\\"MuiTypography-root MuiTypography-body1\\">Статус:</dt>
        </div>
        <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
          <dd class=\\"MuiTypography-root MuiTypography-body1\\">Свободен</dd>
        </div>
      </dl>
      <div class=\\"makeStyles-map-8\\">
        <div data-testid=\\"YMaps\\">
          <div data-testid=\\"Map\\" width=\\"100%\\" height=\\"250\\">
            <div data-testid=\\"Placemark\\" geometry=\\"1,1\\" properties=\\"[object Object]\\" modules=\\"geoObject.addon.balloon\\" options=\\"[object Object]\\"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-lg-9\\">
    <div class=\\"MuiPaper-root makeStyles-paper-6 MuiPaper-elevation1 MuiPaper-rounded\\">
      <div class=\\"makeStyles-root-14\\">
        <div class=\\"makeStyles-title-15\\">
          <h2 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary\\">Доход</h2>
          <p class=\\"MuiTypography-root makeStyles-titleDate-16 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignRight\\"><span>10.12.2020 - 16.12.2020</span></p>
        </div>
        <div data-testid=\\"ResponsiveContainer\\"></div>
      </div>
    </div>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-lg-3\\">
    <div class=\\"MuiPaper-root makeStyles-paper-6 makeStyles-summary-7 MuiPaper-elevation1 MuiPaper-rounded\\">
      <div class=\\"makeStyles-root-17\\">
        <h2 class=\\"MuiTypography-root makeStyles-title-18 MuiTypography-h6 MuiTypography-colorPrimary\\">Сводка</h2>
        <div class=\\"makeStyles-definitionList-19\\">
          <dl class=\\"MuiGrid-root makeStyles-root-11 MuiGrid-container MuiGrid-spacing-xs-2\\">
            <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
              <dt class=\\"MuiTypography-root MuiTypography-body1\\">Доход:</dt>
            </div>
            <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
              <dd class=\\"MuiTypography-root MuiTypography-body1\\">440038</dd>
            </div>
            <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
              <dt class=\\"MuiTypography-root MuiTypography-body1\\">Кол-во Аренд:</dt>
            </div>
            <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
              <dd class=\\"MuiTypography-root MuiTypography-body1\\">203</dd>
            </div>
            <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
              <dt class=\\"MuiTypography-root MuiTypography-body1\\">Происшествий:</dt>
            </div>
            <div class=\\"MuiGrid-root makeStyles-ceil-12 MuiGrid-item MuiGrid-grid-xs-6\\">
              <dd class=\\"MuiTypography-root MuiTypography-body1\\">3</dd>
            </div>
          </dl>
        </div>
        <div class=\\"makeStyles-bottomLink-20\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"/reviews\\">Отзывы</a></div>
      </div>
    </div>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\">
    <div class=\\"MuiPaper-root makeStyles-root-21 MuiPaper-elevation1 MuiPaper-rounded\\">
      <div class=\\"MuiTableContainer-root\\">
        <table class=\\"MuiTable-root makeStyles-root-21\\" aria-label=\\"car list table\\">
          <thead class=\\"MuiTableHead-root\\">
            <tr class=\\"MuiTableRow-root MuiTableRow-head\\">
              <th class=\\"MuiTableCell-root MuiTableCell-head\\" scope=\\"col\\">Дата</th>
              <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\">Доход</th>
              <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\">Кол-во Аренд</th>
              <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\">Происшествий</th>
            </tr>
          </thead>
          <tbody class=\\"MuiTableBody-root\\">
            <tr class=\\"MuiTableRow-root\\">
              <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">10.12.2020</th>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">47822</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">23</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">0</td>
            </tr>
            <tr class=\\"MuiTableRow-root\\">
              <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">11.12.2020</th>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">70306</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">25</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">0</td>
            </tr>
            <tr class=\\"MuiTableRow-root\\">
              <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">12.12.2020</th>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">82181</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">27</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">1</td>
            </tr>
            <tr class=\\"MuiTableRow-root\\">
              <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">13.12.2020</th>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">46865</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">29</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">1</td>
            </tr>
            <tr class=\\"MuiTableRow-root\\">
              <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">14.12.2020</th>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">62191</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">31</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">0</td>
            </tr>
            <tr class=\\"MuiTableRow-root\\">
              <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">15.12.2020</th>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">85703</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">33</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">0</td>
            </tr>
            <tr class=\\"MuiTableRow-root\\">
              <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">16.12.2020</th>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">44970</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">35</td>
              <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">1</td>
            </tr>
          </tbody>
        </table>
      </div>
      <div data-testid=\\"pagination\\"></div>
    </div>
  </div>
</div>"
`;
